lib/ansible/modules/cloud/azure/azure_rm_aci.py

```diff
@@ -0,0 +1,405 @@
+    pass
+
+
+def create_aci_dict(aci):
+    '''
+    Helper method to deserialize a ContainerService to a dict
+    :param: acs: ContainerService or AzureOperationPoller with the Azure callback object
+    :return: dict with the state on Azure
+    '''
+    results = dict(
+        id=aci.id,
+        name=aci.name,
+        tags=aci.tags,
+        location=aci.location,
+        type=aci.type,
+        ip_address=aci.ip_address,
+        restart_policy=aci.restart_policy,
+        state=aci.state,
+        provisioning_state=aci.provisioning_state,
+        image_registry_credentials=aci.image_registry_credentials,
+        volumes=aci.volumes,
+        os_type=aci.os_type
+    )
+    return results
+
+
+class AzureRMContainerInstance(AzureRMModuleBase):
+    """Configuration class for an Azure RM container instance resource"""
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            group_name=dict(
+                type='str',
+                required=True
+            ),
+            image=dict(
+                type='str',
+                required=True
+            ),
+            os=dict(
+                type='str',
+                required=False,
+                default='linux',
+                choices=['linux', 'windows']
+            ),
+            state=dict(
+                type='str',
+                required=False,
+                default='present',
+                choices=['present', 'absent']
+            ),
+            location=dict(
+                type='str',
+                required=False
+            ),
+            service_principal=dict(
+                type='list',
+                required=False
+            )
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.location = None
+        self.tags = None
+        self.state = None
+        self.service_principal = None
+        self.diagnostics_profile = None
+
+        self.results = dict(changed=False, state=dict())
+
+        super(AzureRMContainerInstance, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                       supports_check_mode=True,
+                                                       supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        resource_group = None
+        response = None
+        results = dict()
+        to_be_updated = False
+
+        try:
+            resource_group = self.get_resource_group(self.resource_group)
+        except CloudError:
+            self.fail('resource group {} not found'.format(self.resource_group))
```

Comment:
"resource group {0} not found", keep the code consistency with other string format

Reply:
get_resource_group() has already failed if resource group not found: https://github.com/VSChina/ansible/blob/devel/lib/ansible/module_utils/azure_rm_common.py#L375

Reply:
that comes from ACS implementation, so we should establish what should be right log format and also update other modules if necessary.

Reply:
Yes, I notice that it is come from the ACS module. Regardless the string format, the try/catch seems unnecessary at here, since the `get_resource_group` already panic if there is any exception.